src/driver/sqlite-abstract/AbstractSqliteQueryRunner.ts

```diff
@@ -456,26 +456,34 @@ export abstract class AbstractSqliteQueryRunner extends BaseQueryRunner implemen
     /**
      * Drops the columns in the table.
      */
-    async dropColumns(tableOrName: Table|string, columns: TableColumn[]): Promise<void> {
+    async dropColumns(tableOrName: Table|string, columns: TableColumn[]|string[]): Promise<void> {
         const table = tableOrName instanceof Table ? tableOrName : await this.getCachedTable(tableOrName);
 
         // clone original table and remove column and its constraints from cloned table
         const changedTable = table.clone();
-        columns.forEach(column => {
-            changedTable.removeColumn(column);
-            changedTable.findColumnUniques(column).forEach(unique => changedTable.removeUniqueConstraint(unique));
-            changedTable.findColumnIndices(column).forEach(index => changedTable.removeIndex(index));
-            changedTable.findColumnForeignKeys(column).forEach(fk => changedTable.removeForeignKey(fk));
+        columns.forEach((column: TableColumn|string) => {
+            const columnInstance = column instanceof TableColumn ? column : table.findColumnByName(column);
+            if (!columnInstance)
+                throw new Error(`Column "${column}" was not found in table "${table.name}"`);
+
+            changedTable.removeColumn(columnInstance);
+            changedTable.findColumnUniques(columnInstance).forEach(unique => changedTable.removeUniqueConstraint(unique));
+            changedTable.findColumnIndices(columnInstance).forEach(index => changedTable.removeIndex(index));
+            changedTable.findColumnForeignKeys(columnInstance).forEach(fk => changedTable.removeForeignKey(fk));
         });
 
         await this.recreateTable(changedTable, table);
 
         // remove column and its constraints from original table.
-        columns.forEach(column => {
-            table.removeColumn(column);
-            table.findColumnUniques(column).forEach(unique => table.removeUniqueConstraint(unique));
-            table.findColumnIndices(column).forEach(index => table.removeIndex(index));
-            table.findColumnForeignKeys(column).forEach(fk => table.removeForeignKey(fk));
+        columns.forEach((column: TableColumn|string) => {
+            const columnInstance = column instanceof TableColumn ? column : table.findColumnByName(column);
+            if (!columnInstance)
+                throw new Error(`Column "${column}" was not found in table "${table.name}"`);
+
+            table.removeColumn(columnInstance);
+            table.findColumnUniques(columnInstance).forEach(unique => table.removeUniqueConstraint(unique));
+            table.findColumnIndices(columnInstance).forEach(index => table.removeIndex(index));
+            table.findColumnForeignKeys(columnInstance).forEach(fk => table.removeForeignKey(fk));
         });
```

Comment:
We don't need this block after `await this.recreateTable(changedTable, table);` since `table` is replaced with `changedTable ` in `recreateTable()` method. This is why your test is failing.

Reply:
Thanks @AlexMesser, now all tests passed!